DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/DelegateProgressControllerTests.cs

```diff
@@ -138,6 +141,80 @@ string expectedAction
                 .WithActionName(expectedAction);
         }
 
+        [Test]
+        [TestCaseSource(
+            typeof(DelegateProgressControllerTests),
+            nameof(EditEndpointRedirectTestData)
+        )]
+        public void EditDelegateCourseAdminField_post_calls_service_and_redirects_to_correct_action(
+            DelegateAccessRoute accessedVia,
+            string expectedController,
+            string expectedAction
+        )
+        {
+            // Given
+            const int progressId = 1;
+            const int promptNumber = 1;
+            const string answer = "Test Answer";
+            var formData = new EditDelegateCourseAdminFieldFormData { Answer = answer };
+
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(A<int>._, A<int>._, A<string>._))
+                .DoesNothing();
+
+            // When
+            var result = delegateProgressController.EditDelegateCourseAdminField(
+                formData,
+                promptNumber,
+                progressId,
+                accessedVia,
+                1
+            );
+
+            // Then
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(progressId, promptNumber, answer))
+                .MustHaveHappenedOnceExactly();
+            result.Should().BeRedirectToActionResult().WithControllerName(expectedController)
+                .WithActionName(expectedAction);
+        }
+
+        [Test]
+        public void EditDelegateCourseAdminField_does_not_call_service_with_invalid_model()
+        {
+            // Given
+            const int progressId = 1;
+            const int promptNumber = 1;
+            const string answer = "Test Answer";
+
+            var delegateCourseInfo = new DelegateCourseInfo();
+            var courseAdminField = new CourseAdminFieldWithAnswer(1, "Prompt text", "Answer", null);
+            var courseAdminFields = new List<CourseAdminFieldWithAnswer> { courseAdminField };
+            var attemptStats = new AttemptStats(1, 1);
+            var delegateCourseDetails = new DelegateCourseDetails(delegateCourseInfo, courseAdminFields, attemptStats);
+
+            var formData = new EditDelegateCourseAdminFieldFormData { Answer = answer };
+
+            A.CallTo(() => progressService.UpdateCourseAdminFieldForDelegate(progressId, promptNumber, answer))
+                .DoesNothing();

```

Comment:
Don't think this set up is necessary as we aren't going to be calling this anyway.

Reply:
Oh right, removed :)